src/main/scala/is/hail/expr/AnnotationImpex.scala

```diff
@@ -151,7 +151,7 @@ object SparkAnnotationImpex extends AnnotationImpex[DataType, Any] {
             }.toIndexedSeq
         case TGenotype =>
           val g = a.asInstanceOf[Genotype]
-          Row(g.gt.orNull, g.ad.map(_.toSeq).orNull, g.dp.orNull, g.gq.orNull, g.px.map(_.toSeq).orNull, g.fakeRef, g.isDosage)
+          Row(g.gt.orNull, g.ad.map(_.toSeq).orNull, g.dp.orNull, g.gq.orNull, g.px.map(_.toSeq).orNull, g.fakeRef, g.isLinearScale)
```

Comment:
Actually, it's not _quite_ as bad as I thought. The _.toSeq doesn't add N+3 more allocations as I thought it might.